packages/wdio-appium-service/tests/launcher.test.ts

```diff
@@ -771,6 +774,22 @@ describe('Appium launcher', () => {
                 10000
             )).resolves.toEqual(expect.objectContaining({ spawnargs: expect.arrayContaining(['-e', expect.any(String)]) }))
         })
+
+        test('filter out "Debugger attached" message as an error', async () => {
+            const origSpawn = await vi.importActual<typeof cp>('node:child_process').then((m) => m.spawn)
+            vi.mocked(spawn).mockImplementationOnce(origSpawn)
```

Comment:
I don't think we want to actually start Appium in the unit test. Instead we should mock the `spawn` call and return a mock that allows us to execute the error handler and verify if that error handler works correctly, here is an example:

```ts
const eventListener = { on: vi.fn(), off: vi.fn(), once: vi.fn() }
vi.mocked(spawn).mockReturnValue({
    ...eventListener,
    stdout: { ...eventListener },
    stderr: { ...eventListener },
})

// call function under  test
...

// get error handler
expect(eventListener.on).toBeCalledWith('data', expect.any(Function))
const errorHandler = spawn.mock.calls[0][1].on

// actually call error handler
errorHandler(Buffer.from('some error message'))
expect(mockLogError.error).toBeCalledWith('some error message')
mockLogError.error.mockClear()

errorHandler(Buffer.from('Debugger attached.'))
expect(mockLogError.error).not.toBeCalled()
```

Reply:
I think both serve the same purpose and why isn't launching appium a good idea? I would want an explanation to understand 😅

Reply:
@Delta456 In your unit tests, yes you are starting the node process which simulates the appium behavior. Whereas in @christian-bromann suggestion, he don't start node process and mocks spawn function to return an object with event listeners. This approach directly triggers error handler and test the behavior.

+1 to Christian's suggestion coz its faster and test the error handler functionality directly.

Reply:
@Delta456 unit tests should only focus on testing code and should not have any dependencies to 3rd party resources like an Appium server.

Reply:
@SrinivasanTarget @christian-bromann thanks for the explanations! I have refactored the test.